pkg/api/vlabs/validate_test.go

```diff
@@ -917,7 +917,6 @@ func TestValidateKubernetesLabelKey(t *testing.T) {
 func Test_AadProfile_Validate(t *testing.T) {
 	properties := getK8sDefaultProperties(false)
 	t.Run("Valid aadProfile should pass", func(t *testing.T) {
-		t.Parallel()
```

Comment:
Let's not remove these , unless they really can't be helped. Hopefully having isolated subtests will solve the problem

Reply:
Thanks for the comments @tariq1890 Are you suggesting to update `Test_AadProfile_Validate` into separate subtests? As discussed with @CecileRobertMichon, I noticed  `t.Parallel()` was creating issues where tests in `Test_AadProfile_Validate ` were running in parallel overriding the parameters being tested. Had a hard time reproducing it as it is time sensitive.

Reply:
Oh yes! I see the issue there. Thanks for pointing it out @ritazh . The properties variable was being shared across the parallelized tests within `Test_AadProfile_Validate`.  That should definitely not be the case and I missed it to address that in that particular test method. So the properties variable should be initialized within each subtest in `Test_AadProfile_Validate`. I could push that/ you could make that change in this PR itself.

I was suggesting to create subtests within this method `TestProperties_ValidateZones` so that they can parallelized like the other tests in validate_test.go